src/domains/ecommerce-common/03_storage.tf

```diff
@@ -169,4 +169,421 @@ resource "azurerm_storage_queue" "transactions_dead_letter_queue" {
+  count               = var.env_short == "p" ? 1 : 0
+  name                = "${local.project}-ecommerce-transactions-refund-retry-queue-size-alert"
+  resource_group_name = azurerm_resource_group.storage_ecommerce_rg.name
+  location            = var.location
+
+  action {
+    action_group           = [data.azurerm_monitor_action_group.email.id, data.azurerm_monitor_action_group.slack.id]
+    email_subject          = "Email Header"
+    custom_webhook_payload = "{}"
+  }
+  data_source_id = module.ecommerce_storage_transient.id
+  description    = "Ecommerce transactions refund retry queue size > 10"
+  enabled        = true
+  query = format(<<-QUERY
+    let OpCountForQueue = (operation: string, queueKey: string) {
+      StorageQueueLogs
+      | where OperationName == operation and ObjectKey startswith queueKey
+      | summarize count()
+    };
+    let MessageRateForQueue = (queueKey: string) {
+      OpCountForQueue("PutMessage", queueKey)
+      | join OpCountForQueue("DeleteMessage", queueKey) on count_
+      | project name = queueKey, Count = count_ - count_1
+    };
+    MessageRateForQueue("%s")
+    QUERY
+    , format("%s/pagopa-%s-weu-%s", module.ecommerce_storage_transient.name, var.env_short, "ecommerce-transactions-refund-retry-queue")
+  )
+  severity    = 1
+  frequency   = 15
+  time_window = 30
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 10
+  }
+}
+
+# Ecommerce deadletter queue alert
+resource "azurerm_monitor_diagnostic_setting" "ecommerce_deadletter_queue_diagnostics" {
+  count                      = var.env_short == "p" ? 1 : 0
+  name                       = "${module.ecommerce_storage_deadletter.name}-diagnostics"
+  target_resource_id         = "${module.ecommerce_storage_deadletter.id}/queueServices/default/"
+  log_analytics_workspace_id = data.azurerm_log_analytics_workspace.log_analytics.id
+
+  enabled_log {
+    category = "StorageWrite"
+
+    retention_policy {
+      enabled = true
+      days    = 7
+    }
+  }
+}
+
+# Queue size: Ecommerce - ecommerce notifications service error queue size
+resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_notifications_service_errors_queue" {
+  count               = var.env_short == "p" ? 1 : 0
+  name                = "${local.project}-ecommerce-notifications-service-errors-queue-size-alert"
+  resource_group_name = azurerm_resource_group.storage_ecommerce_rg.name
+  location            = var.location
+
+  action {
+    action_group           = [data.azurerm_monitor_action_group.email.id, data.azurerm_monitor_action_group.slack.id]
+    email_subject          = "Email Header"
+    custom_webhook_payload = "{}"
+  }
+  data_source_id = module.ecommerce_storage_deadletter.id
+  description    = "Ecommerce notifications service error queue size > 1"
+  enabled        = true
+  query = format(<<-QUERY
+      StorageQueueLogs
+      | where OperationName == "PutMessage" and ObjectKey startswith "%s"
+      | summarize count()
```

Comment:
This doesnt' actually count the queue size but only the filling rate (without accounting for message pulls), is it useful to us?

Reply:
This is a deadletter queue. Since it is a dead way for processed event with @infantesimone we decide to trigger an alert if we detect a put in the queue. Of course we can tune this trigger in future, when we will be more confident about our workflow in production.